microsoft-edge/webview2/how-to/add-host-object-to-script.md

```diff
@@ -0,0 +1,295 @@
+
+  * Developers often encounter a native object whose methods or properties they want to call from JavaScript. They may want these native object methods to be triggered by web code or by user interaction on the web side.
+
+  * In addition, developers may not want to re-implement native object functions, and this API enables re-use of code. For example, there might be a webcam native API, which would take a large amount of code to re-write on the web side. Having the ability to call the native object is quicker and more efficient than re-coding the function. In this case you can pass the object to the web side to reuse the native API function.
+
+Scenarios that may benefit from using host objects in script:
+
+  * If there is a keyboard API and the developer wants to call the `keyboardObject.showKeyboard` function from the web side.
+
+  * JavaScript is sandboxed, limiting its ability on the native side. For example, if you need to access a file on the native side, you must use the native file system. If you have a native object exposed to JavaScript via `AddHostObjectToScript`, you can use it to manipulate files on the native file system.
+
+This article uses the WebView2 sample app to demonstrate some practical applications of `AddHostObjectToScript`. For more information about how to embed web content into native applications [Embed web content into native applications - Microsoft Edge Development | Microsoft Docs](/microsoft-edge/webview2/how-to/communicate-btwn-web-native.md).
+
+
+<!-- ====================================================================== -->
+## Before you begin
+
+1. Download and install [Microsoft Visual Studio](https://visualstudio.microsoft.com/) 2019 (16.11.10) or later, and other prerequisites as described in [Get started with WebView2 in Win32 apps - Microsoft Edge Development | Microsoft Docs](/microsoft-edge/webview2/get-started/win32.md).
+
+1. Clone the repo of the sample app from [MicrosoftEdge/WebView2Samples: Microsoft Edge WebView2 samples (github.com)](https://github.com/MicrosoftEdge/WebView2Samples).
+
+1. Run Microsoft Visual Studio and open `WebView2GettingStarted.sln`. Keep the sample app open to follow along with the rest of this article.
+
+Preview of the major steps in this article:
+
+1. Define the host object and implement IDispatch so that WebView2 recognizes the object.
+
+1. Use `AddHostObjectToScript` to pass a method to the web.
+
+1. Call the native object’s methods from web.
+
+
+<!-- ====================================================================== -->
+## Step 1: Define the host object and implement IDispatch
+
+To use this `AddHostObjectToScript` API, you first need to define a host object that implements `IDispatch`. If you already have a host object that implements `IDispatch`, skip to [Step 2: Call the AddHostObjectToScript API](#step-2-call-the-addhostobjecttoscript-api). Implementing `IDispatch` is essentially formatting the host object so that it can be passed to the web.
```

Comment:
```suggestion
To use this `AddHostObjectToScript` API, you first need to define a host object that implements `IDispatch`. If you already have a host object that implements `IDispatch`, skip to [Step 2: Call the AddHostObjectToScript API](#step-2-call-the-addhostobjecttoscript-api). Implementing `IDispatch` is essentially formatting the host object so that it can be passed to the web-side code.
```